app/models/stripe_record.rb

```diff
@@ -76,11 +80,39 @@ def update_status(api_transaction)
   def retrieve_stripe
     case self.stripe_record_type
     when 'payment_intent'
-      Stripe::PaymentIntent.retrieve(self.stripe_id, stripe_account: self.find_account_id)
+      Stripe::PaymentIntent.retrieve(self.stripe_id, stripe_account: self.account_id)
     when 'charge'
-      Stripe::Charge.retrieve(self.stripe_id, stripe_account: self.find_account_id)
+      Stripe::Charge.retrieve(self.stripe_id, stripe_account: self.account_id)
     when 'refund'
-      Stripe::Refund.retrieve(self.stripe_id, stripe_account: self.find_account_id)
+      Stripe::Refund.retrieve(self.stripe_id, stripe_account: self.account_id)
+    end
+  end
+
+  alias_method :retrieve_remote, :retrieve_stripe
+
+  def update_amount_remote(amount_iso, currency_iso)
+    if self.payment_intent?
```

Comment:
- Should we be checking `payment_intent.present?` explicitly instead? 
- Stylistically, do we prefer to explicitly call `self`?

Reply:
Re 1- No, because that's doing different things. In **this** file, there is an `enum` at the top which has a `payment_intent` entry. Declaring this enum also creates the method `payment_intent?` automagically, which is pretty much just a convenience alias for `self.stripe_record_type == 'payment_intent'`

Reply:
Re 2- I do like it because it's more explicit and less "arcane" about where the methods come from. Especially useful in dynamic languages like Ruby where you cannot rely on typing.

Reply:
The point is well-made because I misunderstood what `payment_intent?` was doing here lol.